### ISSUE TITLE
Fresh failures in datalad/git-annex tests -- likely due to new git

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#7116]
Was looking specifically at `test_having_annex` which  fails ATM on Windows and OSX (not Ubuntu). OSX has 2.38.1 git. This seems to be the first time it started to fail and you can see that it has newer git than ubuntu (using 2.30.2 while system has 2.38.1 as well) where it doesn't fail:

```shell
(git)smaug:/mnt/datasets/datalad/ci/git-annex/builds/2022/10[master]git
$> grep -e 'Versions:' -e 'FAILED ' cron-20221022/build-macos.yaml-877-b1412209-failed/1_test-datalad\ \(master\).txt
2022-10-22T07:27:41.2638160Z PASSEDVersions: annexremote=1.6.0 boto=2.49.0 cmd:7z=17.04 cmd:annex=10.20221004-gcde2e6110 cmd:bundled-git=UNKNOWN cmd:git=2.38.1 cmd:ssh=8.1p1 cmd:system-git=2.38.1 cmd:system-ssh=8.1p1 datalad=0.17.7+83.g234632671 humanize=4.4.0 iso8601=1.1.0 keyring=23.9.3 keyrings.alt=UNKNOWN msgpack=1.0.4 patoolib=1.12 platformdirs=2.5.2 requests=2.28.1 tqdm=4.64.1
2022-10-22T07:27:41.2794250Z FAILED ../datalad/tests/test_testrepos.py::test_having_annex - datalad.runner...
2022-10-22T07:27:41.2794760Z FAILED ../datalad/tests/test_testrepos.py::test_clone - datalad.runner.except...
2022-10-22T07:27:41.2795250Z FAILED ../datalad/tests/test_tests_utils_pytest.py::test_with_testrepos - dat...

$> grep -e 'Versions:' -e 'FAILED ' cron-20221022/build-ubuntu.yaml-882-b1412209-success/1_test-datalad\ \(master\).txt 
2022-10-22T05:38:36.8440309Z PASSEDVersions: annexremote=1.6.0 boto=2.49.0 cmd:7z=16.02 cmd:annex=10.20221003+git38-gcde2e6110-1~ndall+1 cmd:bundled-git=2.30.2 cmd:git=2.30.2 cmd:ssh=8.2p1 cmd:system-git=2.38.1 cmd:system-ssh=8.2p1 datalad=0.17.7+83.g234632671 humanize=4.4.0 iso8601=1.1.0 keyring=23.9.3 keyrings.alt=UNKNOWN msgpack=1.0.4 patoolib=1.12 platformdirs=2.5.2 requests=2.28.1 tqdm=4.64.1
```

<details>
<summary>unfortunately I do not see any output associated with that CommandError so impossible to tell what is going on ATM straight from the logs</summary> 

```shell
(git)smaug:/mnt/datasets/datalad/ci/git-annex/builds/2022/10[master]git
$> tail -n 60 cron-20221022/build-macos.yaml-877-b1412209-failed/1_test-datalad\ \(master\).txt
2022-10-22T07:27:41.2781340Z             results.pop('cmd', None)
2022-10-22T07:27:41.2781660Z             results.pop('cwd', None)
2022-10-22T07:27:41.2781890Z             raise CommandError(
2022-10-22T07:27:41.2782150Z                 # whatever the results were, we carry them forward
2022-10-22T07:27:41.2782380Z                 cmd=cmd,
2022-10-22T07:27:41.2782580Z                 cwd=self.cwd,
2022-10-22T07:27:41.2782770Z >               **results,
2022-10-22T07:27:41.2782940Z             )
2022-10-22T07:27:41.2783920Z E           datalad.runner.exception.CommandError: CommandError: 'git clone '/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/datalad_temp_testrepo_1yz21eex/sub dataset1/subm 1' /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/datalad_temp_clone_url_mcpgpcqz' failed with exitcode 128
2022-10-22T07:27:41.2784400Z 
2022-10-22T07:27:41.2784770Z ../../../../hostedtoolcache/Python/3.7.15/x64/lib/python3.7/site-packages/datalad/runner/runner.py:215: CommandError
2022-10-22T07:27:41.2785160Z =============================== warnings summary ===============================
2022-10-22T07:27:41.2785660Z ../../../../hostedtoolcache/Python/3.7.15/x64/lib/python3.7/site-packages/datalad/cli/tests/test_main.py:157
2022-10-22T07:27:41.2786490Z   /Users/runner/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/site-packages/datalad/cli/tests/test_main.py:157: DeprecationWarning: invalid escape sequence \[
2022-10-22T07:27:41.2787100Z     assert re.match("Usage: .*datalad.* \[global-opts\] command \[command-opts\]", stdout)
2022-10-22T07:27:41.2787320Z 
2022-10-22T07:27:41.2787520Z distributed/tests/test_ria_basics.py::test_initremote_basic_httpsurl
2022-10-22T07:27:41.2788000Z tests/test_tests_utils_pytest.py::test_serve_path_via_http[True-None-test1.txt]
2022-10-22T07:27:41.2788520Z tests/test_tests_utils_pytest.py::test_serve_path_via_http[True-None-test_fpath1]
2022-10-22T07:27:41.2789020Z tests/test_tests_utils_pytest.py::test_serve_path_via_http[True-None-test_fpath2]
2022-10-22T07:27:41.2789750Z tests/test_tests_utils_pytest.py::test_serve_path_via_http[True-None-file with space test4]
2022-10-22T07:27:41.2790380Z tests/test_tests_utils_pytest.py::test_serve_path_via_http[True-None-\u0414\u0436\u044d\u0439\u0441\u043e\u043d]
2022-10-22T07:27:41.2790980Z tests/test_tests_utils_pytest.py::test_serve_path_via_http[True-None- |;&%b5{}'"<>\u0394\u05e7\u0645\u0e57\u3042 .datc ]
2022-10-22T07:27:41.2792080Z   /Users/runner/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/site-packages/urllib3/connection.py:465: SubjectAltNameWarning: Certificate for localhost has no `subjectAltName`, falling back to check for a `commonName` for now. This feature is being removed by major browsers and deprecated by RFC 2818. (See https://github.com/urllib3/urllib3/issues/497 for details.)
2022-10-22T07:27:41.2792720Z     SubjectAltNameWarning,
2022-10-22T07:27:41.2792870Z 
2022-10-22T07:27:41.2793220Z -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
2022-10-22T07:27:41.2793760Z =========================== short test summary info ============================
2022-10-22T07:27:41.2794250Z FAILED ../datalad/tests/test_testrepos.py::test_having_annex - datalad.runner...
2022-10-22T07:27:41.2794760Z FAILED ../datalad/tests/test_testrepos.py::test_clone - datalad.runner.except...
2022-10-22T07:27:41.2795250Z FAILED ../datalad/tests/test_tests_utils_pytest.py::test_with_testrepos - dat...
2022-10-22T07:27:41.2795620Z = 3 failed, 1175 passed, 127 skipped, 2 xfailed, 1 xpassed, 8 warnings in 11504.06s (3:11:44) =
2022-10-22T07:27:41.4945660Z ##[error]Process completed with exit code 1.
2022-10-22T07:27:41.5096830Z Post job cleanup.
2022-10-22T07:27:41.6634470Z [command]/Applications/git-annex.app/Contents/MacOS/git version
2022-10-22T07:27:41.7159570Z git version 2.38.1
2022-10-22T07:27:41.7221720Z Copying '/Users/runner/.gitconfig' to '/Users/runner/work/_temp/2016afe0-dbf6-4f00-8978-49880f3538a6/.gitconfig'
2022-10-22T07:27:41.7241250Z Temporarily overriding HOME='/Users/runner/work/_temp/2016afe0-dbf6-4f00-8978-49880f3538a6' before making global git config changes
2022-10-22T07:27:41.7241830Z Adding repository directory to the temporary git global config as a safe directory
2022-10-22T07:27:41.7248970Z [command]/Applications/git-annex.app/Contents/MacOS/git config --global --add safe.directory /Users/runner/work/git-annex/git-annex/datalad
2022-10-22T07:27:41.7931030Z [command]/Applications/git-annex.app/Contents/MacOS/git config --local --name-only --get-regexp core\.sshCommand
2022-10-22T07:27:41.8472320Z [command]/Applications/git-annex.app/Contents/MacOS/git submodule foreach --recursive git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :
2022-10-22T07:27:42.1590180Z [command]/Applications/git-annex.app/Contents/MacOS/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2022-10-22T07:27:42.2089630Z http.https://github.com/.extraheader
2022-10-22T07:27:42.2106050Z [command]/Applications/git-annex.app/Contents/MacOS/git config --local --unset-all http.https://github.com/.extraheader
2022-10-22T07:27:42.2624320Z [command]/Applications/git-annex.app/Contents/MacOS/git submodule foreach --recursive git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :
2022-10-22T07:27:42.6738240Z Post job cleanup.
2022-10-22T07:27:42.8261340Z [command]/Applications/git-annex.app/Contents/MacOS/git version
2022-10-22T07:27:42.8825410Z git version 2.38.1
2022-10-22T07:27:42.8884640Z Copying '/Users/runner/.gitconfig' to '/Users/runner/work/_temp/39dd6c0f-04d0-4784-89f1-30f34ff23449/.gitconfig'
2022-10-22T07:27:42.8904920Z Temporarily overriding HOME='/Users/runner/work/_temp/39dd6c0f-04d0-4784-89f1-30f34ff23449' before making global git config changes
2022-10-22T07:27:42.8905470Z Adding repository directory to the temporary git global config as a safe directory
2022-10-22T07:27:42.8913830Z [command]/Applications/git-annex.app/Contents/MacOS/git config --global --add safe.directory /Users/runner/work/git-annex/git-annex
2022-10-22T07:27:42.9608750Z [command]/Applications/git-annex.app/Contents/MacOS/git config --local --name-only --get-regexp core\.sshCommand
2022-10-22T07:27:43.0148210Z [command]/Applications/git-annex.app/Contents/MacOS/git submodule foreach --recursive git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :
2022-10-22T07:27:43.3262210Z [command]/Applications/git-annex.app/Contents/MacOS/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2022-10-22T07:27:43.3758810Z http.https://github.com/.extraheader
2022-10-22T07:27:43.3775240Z [command]/Applications/git-annex.app/Contents/MacOS/git config --local --unset-all http.https://github.com/.extraheader
2022-10-22T07:27:43.4302550Z [command]/Applications/git-annex.app/Contents/MacOS/git submodule foreach --recursive git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :
2022-10-22T07:27:43.7674790Z Cleaning up orphan processes

```
</details>

edit 1: failed to reproduce locally with only freshier `git`, we started to get it on appveyor for ubuntu, will need to try with recent git annex standalone build.  FTR here is one with new git and older annex, so must be new git related

```
datalad@smaug:/mnt/datasets/datalad/ci/logs/2022/11$ grep -E 'cmd:(annex|git)' 01/pr/7014/b61c0e4/appveyor-8768-failed/245588f9c0je2tvl.txt
[00:07:29]   - cmd:annex: 8.20210903
[00:07:29]   - cmd:git: 2.38.1
```

edit 2: could not reproduce because was using `nodep` (AKA standalone) build of git-annex in conda which bundled older `git` which was used instead. After upgrading `git-annex` to alldep (normal build) which would use "conda-wide" installation of `git` -- reproduced right away. looking into `datalad/tests/test_testrepos.py::test_having_annex`

**Comment by yarikoptic** [comment]
ok, the `transport 'file' not allowed` is the CVE fix in git, more info https://github.blog/2022-10-18-git-security-vulnerabilities-announced/#cve-2022-39253 and found first via https://bugs.launchpad.net/ubuntu/+source/git/+bug/1993586. Adding `protocol.file.allow=always` resolves it, PR is coming (after I see if anything else is needeD)

**Comment by yarikoptic-gitmate** [comment]
Issue fixed in [`0.9.4`](https://github.com/datalad/datalad/releases/tag/0.9.4)

**Comment by yarikoptic-gitmate** [comment]
Issue fixed in [`0.17.9`](https://github.com/datalad/datalad/releases/tag/0.17.9)
